pkg/llmstrategy/react/strategy.go

```diff
@@ -56,41 +56,45 @@ type Strategy struct {
 
 type Conversation struct {
 	strategy *Strategy
-	workDir  string
 
-	// recorder captures events for diagnostics
 	recorder journal.Recorder
 
-	userInterface ui.UI
-
-	llmClient gollm.Client
+	UI      ui.UI
+	llmChat gollm.Chat
 
 	MaxIterations int
 
-	previousQueries  []string
-	currentIteration int
-	messages         []Message
+	workDir string
 }
 
 func (s *Strategy) NewConversation(ctx context.Context, userInterface ui.UI) (llmstrategy.Conversation, error) {
 	log := klog.FromContext(ctx)
 
-	// Create a temporary working directory
 	// Create a temporary working directory
 	workDir, err := os.MkdirTemp("", "agent-workdir-*")
 	if err != nil {
 		return nil, fmt.Errorf("creating temporary working directory: %w", err)
 	}
 
 	log.Info("Created temporary working directory", "workDir", workDir)
+	data := PromptData{
+		Tools: s.Tools,
+	}
+
+	systemPrompt, err := s.generatePrompt(ctx, defaultSystemPromptTemplate, data)
+	if err != nil {
+		log.Error(err, "Failed to generate system prompt")
```

Comment:
Nit: Why this pattern instead of just `return fmt.Errorf("failed to generate system prompt: %w", err)`